sklearn/utils/_metadata_requests.py

```diff
@@ -987,7 +1022,7 @@ def _route_params(self, *, params, method):
         res.update(child_params)
         return res
 
-    def route_params(self, *, caller, params):
+    def route_params(self, *, caller, params, parent=None):
```

Comment:
I don't think we should have this default then, the user should always pass. If it's none, we'll fail.

Reply:
Alright, and I have adjusted the places where it was used in the code, especially where the scorer functions are tested.